File: 3817-maximum-containers-on-a-ship/3817-maximum-containers-on-a-ship.py
class Solution:
    def maxContainers(self, n: int, w: int, maxW: int) -> int:

        maxAcceptableWight = maxW // w
        cells = n ** 2

        if maxAcceptableWight > cells:
            return cells
        elif maxAcceptableWight < cells:
            return maxAcceptableWight

        return cells

File: 3817-maximum-containers-on-a-ship/3817-maximum-containers-on-a-ship.cpp
class Solution {
public:
    int maxContainers(int n, int w, int maxW) {
        int maxAc = maxW / w;
        int cells = n * n;
        if (maxAc < cells) {
            return maxAc;
        } else if (maxAc > cells) {
            return cells;
        }

        return cells;
    }
};